### PR TITLE
Upgrade Infinispan to 13.0.5.Final

### PR DESCRIPTION
**Pull request by karesti**

### CODE REVIEW

File: extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServicesConfig.java [hunk 1]
```diff
@@ -54,6 +56,14 @@
     @ConfigItem(defaultValue = "infinispan")
     public String serviceName;
 
+    /**
+     * List of the artifacts to download and be present in the server lib. For example a maven coordinate
+     * (org.postgresql:postgresql:42.3.1) or dependency location url.
```
Comment (geoand) [inline extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServicesConfig.java:61]: I personally don't understand what  this does (and I assume users won't either), so maybe expand on this a little more?

Furthermore, what happens if the string is invalid? What kind of error does Infinispan display?

File: extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServiceProcessor.java [hunk 2]
```diff
@@ -208,6 +210,7 @@ public QuarkusInfinispanContainer(OptionalInt fixedExposedPort, String serviceNa
             }
             withUser(DEFAULT_USERNAME);
             withPassword(InfinispanDevServiceProcessor.DEFAULT_PASSWORD);
+            artifacts.map(a -> withArtifacts(a.toArray(new String[0])));
```
Comment (wburns) [inline extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServiceProcessor.java:213]:
```suggestion
            artifacts.ifPresent(a -> withArtifacts(a.toArray(new String[0])));
```

File: extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServicesConfig.java (hunk 1 above)
Comment (karesti) [reply extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServicesConfig.java:61]: The container won't start if you ask for an invalid maven dependency. This is useful when using persistence with caches IllegalArgumentException: Artifact 'org.postgresql:postgresql:jjj' not found

Comment (geoand) [reply extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServicesConfig.java:61]: Sure, but I think the javadoc should reflect this

Comment (gsmet) [reply extensions/infinispan-client/deployment/src/main/java/io/quarkus/infinispan/client/deployment/devservices/InfinispanDevServicesConfig.java]: I made an adjustment and force pushed in an attempt to get this in 2.6.1.Final.

**Review by geoand — APPROVED** [review]

**Pull request merged**